kirill_tolstobrov/logger.cpp

```diff
@@ -5,15 +5,11 @@
 
 namespace uni_cpp_practice {
 void Logger::set_file(const std::optional<std::string>& filename) {
-  if (file_stream_.has_value()) {
+  if (file_stream_->is_open()) {
     file_stream_->close();
     file_stream_ = std::nullopt;
   }
```

Comment:
Зачем убрал проверку на `filename`?
```suggestion
  if (!filename.has_value()) {
    file_stream_ = std::nullopt;
    return;
  }
```